src/main/java/net/floodlightcontroller/staticflowentry/StaticFlowEntryPusher.java

```diff
@@ -365,19 +365,21 @@ void parseRow(Map<String, Object> row, Map<String, Map<String, OFFlowMod>> entri
 				if (row.get(key) == null) {
 					continue;
 				}
+				
 				if (key.equals(COLUMN_SWITCH) || key.equals(COLUMN_NAME) || key.equals("id")) {
 					continue; // already handled
 				}
-				// explicitly ignore timeouts and wildcards
-				if (key.equals(COLUMN_HARD_TIMEOUT) || key.equals(COLUMN_IDLE_TIMEOUT)) {
```

Comment:
Yeah, that makes sense. We wanted to include the ability to do this though.
Many have asked for it in the past, so I figured I'd at least make it an
option. I had intended to do that for the release, but somehow it got
reverted when I did the merge and I didn't realize it. There will be a note
in the SFP documentation that this feature may not be stable.

Ryan Izard
Graduate Research/Teaching Assistant
306B Fluor Daniel Building
ECE Department, Clemson University
Clemson, SC 29634
rizard@g.clemson.edu
---------------------------------------------------
Big Switch Networks
ryan.izard@bigswitch.com

On Wed, Feb 4, 2015 at 1:50 PM, Jason Parraga <notifications@github.com>
wrote:

> In
> src/main/java/net/floodlightcontroller/staticflowentry/StaticFlowEntryPusher.java
> <https://github.com/floodlight/floodlight/pull/491#discussion_r24108112>:
>
> >  				if (key.equals(COLUMN_SWITCH) || key.equals(COLUMN_NAME) || key.equals("id")) {
> >  					continue; // already handled
> >  				}
> > -				// explicitly ignore timeouts and wildcards
> > -				if (key.equals(COLUMN_HARD_TIMEOUT) || key.equals(COLUMN_IDLE_TIMEOUT)) {
>
> So the notion of the static flow pusher is that it pushes static STATIC
> flows. If you respect the hard/idle timeout values flows can expire, but
> will still be displayed by the SFP database which is inconsistent. Make
> sense? @rizard <https://github.com/rizard>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/floodlight/floodlight/pull/491/files#r24108112>.
>